### PR TITLE
PLF-8447 : "Send me a digest email" on notification settings is not working

### PR DESCRIPTION
**Pull request by rdenarie**

### CODE REVIEW

**Comment by rdenarie** [comment]
Comments from initial PR : https://github.com/exodev/platform/pull/409
- Avoid using DAO and use storage layer instead :ok: 
- We shouldn't delete all digests only when the "old version" is greater than 5.2.0 (the version on which the bug happened) :ok: 
Test made with migration from 5.2.x->6.0.x : digest correctly deleted.
Test made with migration from 5.1.x->6.0.x : digest not deleted.
Test made with migration from 5.1.x->5.2.x->6.0.x : digest not deleted during first step, UP ignored during step
- if an exception happens, it should be thrown to be sure that the plugin is re-executed until it succeeds :ok:

cc @boubaker

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        //force remove daily notification digest
+        context.append(NotificationJob.JOB_WEEKLY, false);
+        context.append(NotificationJob.JOB_DAILY, true);
+        mailNotificationStorage.removeMessageAfterSent(context);
+      }
+
+      schedulerService.resumeJob("NotificationDailyJob", "Notification");
+      schedulerService.resumeJob("NotificationWeeklyJob", "Notification");
+    } catch (RuntimeException e) {
+      throw e;
```
Comment (boubaker) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java]: no need to catch the exception, you can use try / finally block without a catch block

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+
+
+    try {
+      //Force remove digest items only source migration version is after 5.2.0
+      //Before this version, there is no problem of blocking digest, so no need of delete it
+      if (VersionComparator.isAfter(oldVersion, "5.2.0")) {
```
Comment (boubaker) [inline component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java]: if old version >= 5.2.0 => Delete queue messages

`VersionComparator.isAfter(oldVersion, "5.2.0") || VersionComparator.isSame(oldVersion, "5.2.0")`

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java (hunk 1 above)
Comment (rdenarie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java]: No, I cant remove the catch :
image
Error : Unandled Exception

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java (hunk 2 above)
Comment (rdenarie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java]: done

File: component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java (hunk 1 above)
Comment (boubaker) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java]: ok, at least delete the catch of [unchecked exception](https://crunchify.com/better-understanding-on-checked-vs-unchecked-exceptions-how-to-handle-exception-better-way-in-java/) (RuntimeException), the next block is sufficient to manage all exception, except those of type `Error` of course.

Comment (rdenarie) [reply component/upgrade/plugins/src/main/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePlugin.java]: done

File: component/upgrade/plugins/src/test/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePluginTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.exoplatform.commons.api.notification.service.storage.MailNotificationStorage;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.scheduler.JobSchedulerService;
+import org.junit.Test;
+
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Created by Romain Dénarié (romain.denarie@exoplatform.com) on 07/11/19.
```
Comment (boubaker) [inline component/upgrade/plugins/src/test/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePluginTest.java]: This kind of header is not accepted anymore. Please use class header to describe what is this class for.

**Comment by rdenarie** [comment]
Done for the unit test

File: component/upgrade/plugins/src/test/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePluginTest.java (hunk 3 above)
Comment (rdenarie) [reply component/upgrade/plugins/src/test/java/org/exoplatform/platform/upgrade/plugins/ResumeDigestJobUpgradePluginTest.java]: done

**Pull request merged**